lms/data_tasks/report/create_from_scratch/01_functions/01_date_functions.sql

```diff
@@ -11,7 +11,18 @@ IMMUTABLE
 AS $$
     SELECT CASE
         WHEN resolution = 'all_time' THEN '1901-01-01'::DATE
-        WHEN resolution = 'academic_year' THEN (DATE_TRUNC('year', date - INTERVAL '6 month') + INTERVAL '6 month')::DATE
+        WHEN resolution = 'academic_year' THEN (
+            DATE_TRUNC('year', date - INTERVAL '6 month') + INTERVAL '6 month'
+        )::DATE
+        -- Trailing year is a time period based on "today" as the end-period
+        -- quantized by year.
+        WHEN resolution = 'trailing_year' THEN (
+            -- Offset the date by the amount left when we snip the year off
+            -- from now and truncate to the year start
+            DATE_TRUNC('year', date - (CURRENT_DATE - DATE_TRUNC('year',CURRENT_DATE)))
+            -- Add back on the amount left when we snip the year off from "now"
+            + (CURRENT_DATE - DATE_TRUNC('year', CURRENT_DATE))
+        )::DATE
```

Comment:
I can't help but think there's a simplification we can here, but I can't spot it.